json-rpc/src/methods.rs

```diff
@@ -80,6 +80,72 @@ async fn get_metadata(service: JsonRpcService, _: Vec<Value>) -> Result<BlockMet
     Ok(BlockMetadata { version, timestamp })
 }
 
+/// Returns transactions by range
+async fn get_transactions(
+    service: JsonRpcService,
+    params: Vec<Value>,
+) -> Result<Vec<TransactionView>> {
+    let p_start_version: Number = serde_json::from_value(params[0].clone())?;
+    let p_limit: Number = serde_json::from_value(params[1].clone())?;
+    let include_events: bool = serde_json::from_value(params[2].clone())?;
+
+    let start_version = p_start_version
+        .as_u64()
+        .ok_or(format_err!("Invalid start_version"))?;
+    let limit = p_limit.as_u64().ok_or(format_err!("Invalid limit"))?;
+
+    ensure!(
```

Comment:
I wonder, isn't it checked down the stack in libradb itself

Reply:
I think it shouldn't be, we should put the limit on the API layer, not the storage layer.